packages/office-ui-fabric-react/src/components/Icon/examples/Icon.RegisterIconRenderer.Example.tsx

```diff
@@ -0,0 +1,70 @@
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+import { Icon, IconType, IIconProps, registerIconRenderer } from 'office-ui-fabric-react/lib/Icon';
+import './IconExample.scss';
+
+export class IconRegisterIconRendererExample extends React.Component<any, any> {
+  private _unsubscribeRendererFn: () => void;
+
+  public componentWillMount() {
+    this._unsubscribeRendererFn = registerIconRenderer((iconProps: IIconProps) => {
+      // This function can return any valid JSX.Element icon (e.g. an svg, an svg use reference, an img that points to a png/bmp, a sprite w/ appropriate css styling applied, etc.)
+      if (iconProps.iconName === 'Dot') {
+        return (
+          <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 100 100'>
+            <circle cx='50' cy='50' r='40' fill='orange' />
+          </svg>);
+      }
+      return iconProps.iconName === 'OneDrive' ? oneDriveIcon() :
+        iconProps.iconName === 'Yammer' ? yammerIcon() :
+          null;
+    });
+  }
+
+  public componentWillUnmount() {
+    this._unsubscribeRendererFn && this._unsubscribeRendererFn();
+  }
+
+  public render() {
+    return (
+      <div>
+        <Icon
+          iconName={ 'Dot' }
+          iconType={ IconType.image }
+          className={ 'ms-IconRegisterIconRendererExample-dot' }
+        />
+        <Icon
+          iconName={ 'OneDrive' }
+          iconType={ IconType.image }
```

Comment:
Do we think we should have an icontype.svg?

Reply:
That way we could only do the render function when it is an SVG and fall back to the fabric default otherwise?

Reply:
but you don't necessarily want to return an SVG from this function

Reply:
So there is image and default (font) for icontype. Using image to specify svg seems a bit strange to me.

Reply:
When you are defining an icon, I think it is reasonable to say I want the 'OneDrive svg'

Reply:
And then this function is responsible for figuring out how to get it (inline vs from a sprite, etc)

Reply:
I considered adding an iconType called custom (or something like that) but don't think that we want to call it SVG since this is more general purpose (e.g. a runtime generated png icon).  In the end I didn't think the custom iconType added anything beyond the existing image iconType

Reply:
Fair point, how do you envision us determining in office online that we should use a render function instead of the default?

Reply:
Consumers will decide based on iconProps (likely iconName specifically) whether to render using the function... Office Online will likely render all of its icons using the function (ignoring any components that we may consume but not author).